src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayViewController.swift

```diff
@@ -0,0 +1,174 @@
+				self?.tableView.scrollToRow(at: IndexPath(row: 0, section: 0), at: .top, animated: false)
+				self?.updateForSelectedEntryType()
+			}
+			.store(in: &subscriptions)
+	}
+
+	// MARK: - Protocol UITableViewDataSource
+
+	func numberOfSections(in tableView: UITableView) -> Int {
+		return 2
+	}
+
+	func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		switch section {
+		case 0:
+			return 1
+		case 1:
+			return viewModel.entriesOfSelectedType.count
+		default:
+			fatalError("Invalid section")
+		}
+	}
+
+	func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		switch indexPath.section {
+		case 0:
+			return addCell(forRowAt: indexPath)
+		case 1:
+			return entryCell(forRowAt: indexPath)
+		default:
+			fatalError("Invalid section")
+		}
+	}
+
+	// MARK: - Protocol UITableViewDelegate
+
+	func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		switch indexPath.section {
+		case 0:
+			onAddEntryCellTap(viewModel.day, viewModel.selectedEntryType)
+		case 1:
+			viewModel.toggleSelection(of: viewModel.entriesOfSelectedType[indexPath.row])
+		default:
+			fatalError("Invalid section")
+		}
+	}
+
+	// MARK: - Private
+
+	private let viewModel: DiaryDayViewModel
+	private let onAddEntryCellTap: (DiaryDay, DiaryEntryType) -> Void
+
+	private var subscriptions = [AnyCancellable]()
+
+	@IBOutlet weak var segmentedControl: UISegmentedControl!
+	@IBOutlet weak var tableView: UITableView!
+
+	private func setupSegmentedControl() {
+		segmentedControl.setTitleTextAttributes([NSAttributedString.Key.font: UIFont.enaFont(for: .subheadline)], for: .normal)
+		segmentedControl.setTitleTextAttributes([NSAttributedString.Key.font: UIFont.enaFont(for: .subheadline, weight: .bold)], for: .selected)
+
+		segmentedControl.setTitle(AppStrings.ContactDiary.Day.contactPersonsSegment, forSegmentAt: 0)
+		segmentedControl.setTitle(AppStrings.ContactDiary.Day.locationsSegment, forSegmentAt: 1)
+	}
+
+	private func setupTableView() {
+		tableView.register(
+			UINib(nibName: String(describing: DiaryDayAddTableViewCell.self), bundle: nil),
+			forCellReuseIdentifier: String(describing: DiaryDayAddTableViewCell.self)
+		)
+
+		tableView.register(
+			UINib(nibName: String(describing: DiaryDayEntryTableViewCell.self), bundle: nil),
+			forCellReuseIdentifier: String(describing: DiaryDayEntryTableViewCell.self)
+		)
+
+		tableView.delegate = self
+		tableView.dataSource = self
+
+		tableView.separatorStyle = .none
+		tableView.rowHeight = UITableView.automaticDimension
+		tableView.estimatedRowHeight = 60
+	}
+
+	private func addCell(forRowAt indexPath: IndexPath) -> UITableViewCell {
+		guard let cell = tableView.dequeueReusableCell(withIdentifier: String(describing: DiaryDayAddTableViewCell.self), for: indexPath) as? DiaryDayAddTableViewCell else {
+			fatalError("Could not dequeue DiaryDayAddTableViewCell")
```

Comment:
Hm, if we get there, it's a programming error. And it's broken either way, either we drown the support hotline because the diary is empty or because it crashes the app. I think crashing at least doesn't suggest to the user that any data got lost. So I would like to leave the fatal error in. Let's add UI tests in the end that guarantee that this works as expected.